### PR TITLE
[WETH test] Add Reentrancy test

### PR DESCRIPTION
**Pull request by jaglinux**
Possible attack vector:
_burn is called after eth transfer in WETH withdraw().
In Erc20 _burn(), if the code "balance[from] -=amount"
is unchecked or older solidity compiler version (0.6 or lesser) is used
which does not check for arithmetic underflow.

Signed-off-by: Jagadish Krishnamoorthy <jagdish.krishna@gmail.com>

## Description

Describe the changes made in your pull request here.

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [ ] Ran `forge snapshot`?
- [ ] Ran `npm run lint`?
- [ ] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

File: src/test/utils/MockReentrancyHack.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        weth.deposit{value: 1 ether}();
+        weth.withdraw(1 ether);
+    }
+
+    receive() external payable {
+        if (address(weth).balance >= 1 ether) {
+            weth.withdraw(1 ether);
+        }
+    }
+}
```
Comment (transmissions11) [inline src/test/utils/MockReentrancyHack.sol:28]: nice! can you just paste this at the top of WETH.t.sol?

File: src/test/WETH.t.sol [hunk 2]
```diff
@@ -104,6 +105,17 @@ contract WETHTest is DSTestPlus {
         assertEq(weth.totalSupply(), depositAmount - withdrawAmount);
     }
 
+    function testReentrancy() public {
```
Comment (transmissions11) [inline src/test/WETH.t.sol:108]: can we name this something like testReentrancyFails

**Comment by transmissions11** [comment]
nice work, always good to test edge cases

File: src/test/WETH.t.sol (hunk 2 above)
Comment (jaglinux) [reply src/test/WETH.t.sol]: done

File: src/test/utils/MockReentrancyHack.sol (hunk 1 above)
Comment (jaglinux) [reply src/test/utils/MockReentrancyHack.sol]: done

**Comment by jaglinux** [comment]
@transmissions11 any updates ?
